.github/workflows/plugins_react_tests.yml

```
@@ -13,27 +13,43 @@ permissions:
   contents: read
 
 jobs:
+  setup_matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        run: |
+          curl --silent --show-error --fail --output matrix.json https://raw.githubusercontent.com/theforeman/foreman/${{ github.base_ref }}/.github/matrix.json
+          cat >matrix_include.json <<EOF
+          [{"repo": "foreman-tasks", "org": "theforeman"}, {"repo": "katello", "org": "Katello"}]
+          EOF
+          {
+            echo 'matrix<<EOF'
+            python3 -c 'import json; matrix=json.load(open("matrix.json")); matrix["include"]=json.load(open("matrix_include.json")); print(json.dumps(matrix));'
+            echo 'EOF'
+          } >> "$GITHUB_OUTPUT"
```

Similar to https://github.com/theforeman/foreman/pull/9880#discussion_r1374478530 I think we should test it from the ref itself instead of the base ref.

And use the composite action
```suggestion
        uses: ekohl/gha-matrix-builder@v0
        with:
          base_matrix_url: "https://raw.githubusercontent.com/${{ github.repository }}/${{ github.ref_name }}/.github/matrix.json"
          include: '[{"repo": "foreman-tasks", "org": "theforeman"}, {"repo": "katello", "org": "Katello"}]'
```